src/TabSwipe/widget/TabSwipe.ts

```diff
@@ -0,0 +1,174 @@
+import * as dojoDeclare from "dojo/_base/declare";
+import * as WidgetBase from "mxui/widget/_WidgetBase";
+import * as aspect from "dojo/aspect";
+import * as domConstruct from "dojo/dom-construct";
+import * as registry from "dijit/registry";
+
+import { SwipeHandler } from "./SwipeHandler";
+import "./ui/TabSwipe.scss";
+
+export interface TabContainer extends mxui.widget._WidgetBase {
+    declaredClass: "mxui.widget.TabContainer";
+    showTab: (tab: TabPane) => void;
+    focusIndex: number;
+    onShowTab: (callback: () => void ) => void;
+    onHideTab: (callback: () => void ) => void;
+    tabSwipeId: string;
+    validator: null;
+    _active: TabPane;
+    _tabList: HTMLElement;
+    _tabPanes: TabPane[];
+    _tabContent: TabContent;
+    _clickHandler: null;
+}
+
+export interface TabContent extends HTMLElement {
+    declaredClass: "mxui.widget.TabContent";
+}
+
+export interface TabPane extends mxui.widget._WidgetBase {
+    container: TabContainer;
+    _hidden: boolean;
+    showTab: (tab: TabPane) => void;
+    show: () => void;
+    hideTab: () => void;
+    index: number;
+    _visible: boolean;
+    visibilityIndex: number;
+}
+
+class TabSwipe extends WidgetBase {
+    private targetName: string;
+    private tabNavStyle: "tabs"| "indicators";
+    private lazyLoad: boolean;
```

Comment:
the class members that are part of the widget.xml are not private but public. (Technically they are mixed in)